### DISCUSSION TITLE
Jetson Orin AGX Development kit - Booting from SD card

### DISCUSSION THREAD

**Discussion by eligavril** [OE4T/meta-tegra#1221] [Q&A]
Hi,

I'm just getting started with Jetson Orin AGX dev kit, and my current goal is to get it to boot from SD card. I managed to do the same with Jetson Nano using the guide in [here](https://github.com/OE4T/meta-tegra/wiki/Flashing-the-Jetson-Dev-Kit) but unfortunately I can't find any straightforward guide for Orin AGX. In fact, it isn't even clear to me if this is possible. The guide I used for Nano doesn't seem to work and Orin AGX isn't listed there as one of the kits supporting SD card boot.

Could someone help me out? I'm still relatively new to both Jetson and Yocto.

**Comment by dwalkes** [comment]
Based on [this link](https://docs.nvidia.com/jetson/archives/r35.3.1/DeveloperGuide/text/SD/FlashingSupport.html#flashing-to-an-sd-card) it doesn't look like sdcard boot is supported by NVIDIA on Orin AGX.

Have you noticed the instructions at https://github.com/OE4T/meta-tegra/wiki/initrd-flashing-support ?  This should allow you to boot from USB instead I believe.

**Reply by eligavril** [reply]
Hi,

Thanks for the reply! I'm trying these instructions now. I've tried to execute the initrd-flash script through both WSL and a Linux VM with no luck. It keeps getting stuck on "waiting for USB storage device flashpkg" because the USB storage device never shows up. Any chance to get it working or do I need a native linux machine?

Also, could I perhaps use this method to boot from SD card? Will it work if I just put the name of the SD card device in TNSPEC_BOOTDEV variable?

**Reply by madisongh** [reply]
You can try it.  Whether the UEFI bootloader supports booting off the SDcard on the AGX Orin or not, I'm not sure.  When I get a chance, I'll test booting from an SDcard myself, and if I can get it to work, I'll update the tegraflash packager to include a `dosdcard.sh` script.

**Reply by dwalkes** [reply]
>  I've tried to execute the initrd-flash script through both WSL and a Linux VM with no luck. It keeps getting stuck on "waiting for USB storage device flashpkg" because the USB storage device never shows up. Any chance to get it working or do I need a native linux machine?

I think you'll eventually want a dedicated Ubuntu machine if you are going to be working with Jetson stuff.  You are fighthing an uphill battle with anything else.

**Reply by eligavril** [reply]
Well, I'm apparently having more basic issues. Even if I'm not trying to boot from SD I can't get it to work. I followed the instructions in tegra-demo-distro (jetson-agx-orin-devkit, kirkstone branch) and managed to flash the device. I connected the device to a screen via display-port-to-hdmi converter and I could indeed see the UEFI bootloader. However, it said it couldn't find extlinux.conf. Then the screen stops receiving signals. I tried using my own custom machine without tegra-demo and the same thing happened. I saw that there was a variable called UBOOT_EXTLINUX which  needs to be set to 1 in case of extlinux. I did that and the error stopped, but after UEFI I still didn't see anything (I tried that with a custom machine and the core-image-sato image). I also tried adding "l4t-usb-device-mode" to see if I'm getting any output from USB but no device appeared on my computer. To check if the flash worked, I patched the source code of UEFI to print some stuff and I saw the prints. Which means I am flashing correctly (at least the bootloader).

Any ideas?

**Reply by dwalkes** [reply]
I don't have an agx orin devkit so I can't try that specific combination but based on https://docs.google.com/spreadsheets/d/1EWG5QZvtWdk74gz48D4xCbegK-bsqMFJ8XOsyGdUUeI/edit#gid=1125662453 it looks like the master branch `demo-image-full` build has been tested successfully if you are just looking for something which boots.

**Reply by eligavril** [reply]
small update - My device actually does boot under the kirkstone branch. I can access it via micro-USB serial console. I'll experiment with booting from SD card and update if there is anything interesting. Thanks for the help! This yocto layer for nvidia jetson is amazing!

**Reply by Deepak-Beniya** [reply]
@eligavril 
Could you help me to build the yocto source for Jetson orin devkit board. 
and I also want to boot from sd card . 
Is there any one single file (like .sdcard image) it should include primary ,secondary bootloader, kernel,zImage,rootfs . so, that i will send it to my friend.
If you have any log file that would be greatfull for me .
